src/react/projects/spark-core-react/src/SprkFooter/components/SprkFooterConnectIcons/SprkFooterConnectIcons.js

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import { uniqueId } from 'lodash';
+import PropTypes from 'prop-types';
+import SprkIcon from '../../../SprkIcon/SprkIcon';
+
+class SprkFooterConnectIcons extends Component {
+  constructor(props) {
+    super(props);
+
+    if (props.connectIcons.icons !== undefined) {
+      this.state = {
+        connectIconsHasIds: props.connectIcons.icons.map(
+          icon => ({ id: uniqueId(), ...icon }),
+        ),
+      };
+    }
+  }
+
+  render() {
+    const { connectIcons } = this.props;
+    if (connectIcons.icons === undefined) return false;
+    const { connectIconsHasIds } = this.state;
+    return (
+      <div className="sprk-o-Stack__item sprk-o-Stack sprk-o-Stack--large sprk-o-Box">
+        <h3 className="sprk-o-Stack__item sprk-b-TypeBodyOne">
+          {connectIcons.heading}
+        </h3>
+
+        <div className="sprk-o-Stack__item sprk-o-Stack sprk-o-Stack--large sprk-o-Stack--split@m">
+          <ul className="sprk-o-Stack__item sprk-o-Stack__item--flex@m sprk-o-Stack sprk-o-Stack--split@xxs sprk-o-Stack--medium sprk-b-List sprk-b-List--bare">
+            {connectIconsHasIds.map((icon) => {
+              const TagName = icon.linkElement || 'a';
+              return (
+                <li key={icon.id} className="sprk-o-Stack__item">
+                  <TagName
+                    className="sprk-b-Link sprk-b-Link--plain"
+                    href={TagName === 'a' ? icon.href || '#nogo' : undefined}
+                    data-analytics={icon.analyticsString}
+                  >
+                    <SprkIcon iconName={icon.name} additionalClasses={`sprk-c-Icon--stroke-current-color sprk-c-Icon--l ${icon.addClasses}`} />
+                    <span className="sprk-u-ScreenReaderText">{icon.screenReaderText}</span>
+                  </TagName>
+                </li>
+              );
+            })}
+          </ul>
+        </div>
+      </div>
+    );
+  }
+}
+
+SprkFooterConnectIcons.defaultProps = {
+  connectIcons: {},
```

Comment:
should include connectIcons.icons with a default value, or make connectIcons.icons.isRequired

Reply:
The whole connectIcons section is optional.

Reply:
But should it be for the SprkFooterConnectIcons component?

Reply:
I agree! Fixing now!